test_pull_requests

```diff
@@ -499,192 +486,118 @@ end
-            res.error!
-          end
-        end
+        auto_paginate = true
+        commits = pull_commits("#{@properties['github_user']}/#{repo}", id)
+        auto_paginate = false
       end
       commits.last
     end
 
-    def get_comments(issue_id, repo, num_tries=3)
-      page = 1
-      comments = []
-      last_comments = nil
-      while (!last_comments || last_comments.length == 100)
-        (1..num_tries).each do |i|
-          res = get "#{GITHUB_API_BASE_URL}/repos/#{@properties['github_user']}/#{repo}/issues/%s/comments?page=#{page}&per_page=100" % issue_id
-          if res.success?
-            last_comments = res.data
-            comments += last_comments
-            page += 1
-            break
-          end
-          res.error! if i == num_tries
-        end
-      end
-      comments
+    def get_comments(issue_id, repo)
+      auto_paginate = true
+      pull_comments("#{@properties['github_user']}/#{repo}", issue_id)
+    ensure
+      auto_paginate = false
     end
 
-    def delete_comment(comment_id, repo, num_tries=3)
+    def delete_comment(comment_id, repo)
       $stderr.puts "  #{@dry_run_msg}Deleting comment ##{comment_id}"
-      (1..num_tries).each do |i|
-        res = delete "#{GITHUB_API_BASE_URL}/repos/#{@properties['github_user']}/#{repo}/issues/comments/%s" % [comment_id]
-        # If we get 404 on a retry, that means there's no comment to delete
-        break if res.success? || (res.code == '404' && i > 1)
-        res.error! if i == num_tries
-      end if !@dry_run
+      delete_pull_comment("#{@properties['github_user']}/#{repo}", comment_id) if !@dry_run
     end
 
-    def add_comment(issue_id, repo, comment, num_tries=3)
+    def add_comment(issue_id, repo, comment)
       if @dry_run
-        $stderr.puts "  #{@dry_run_msg}add_comment"
+        $stderr.puts "  Dry run; skipping: add_comment"
         return {'id' => 0, 'body' => comment, 'updated_at' => Time.now}
-      end
-      params = { :body => comment }
-      (1..num_tries).each do |i|
-        res = post "#{GITHUB_API_BASE_URL}/repos/#{@properties['github_user']}/#{repo}/issues/%s/comments" % [issue_id], params
-        return res.data if res.success?
-        res.error! if i == num_tries
+      else
+        add_comment("#{@properties['github_user']}/#{repo}", issue_id, comment)
       end
     end
 
-    def update_comment(comment_id, repo, comment, num_tries=3)
+    def update_comment(comment_id, repo, comment)
       $stderr.puts "  #{@dry_run_msg}Updating comment ##{comment_id} with #{comment}"
-      params = { :body => comment }
-      (1..num_tries).each do |i|
-        res = post "#{GITHUB_API_BASE_URL}/repos/#{@properties['github_user']}/#{repo}/issues/comments/%s" % [comment_id], params
-        return res.data if res.success?
-        res.error! if i == num_tries
-      end if !@dry_run # We don't use the result
+      gc.update_comment("#{@properties['github_user']}/#{repo}", comment_id, comment) if !@dry_run
```

Comment:
Wait, but here we need to be clear it's the `gc.update_comment()`? Unclear if we need to always do this or not but it may be nice to always do `gc.<whatever>` just for clarity's sake